enhancements/microshift/etcd-as-a-transient-systemd-unit.md

```diff
@@ -0,0 +1,221 @@
+---
+title: etcd-as-a-transient-systemd-unit
+authors:
+  - dusk125
+  - hasbro17
+reviewers:
+  - "@tjungblu, etcd Team"
+  - "@Elbehery, etcd Team"
+  - "@EmilyM1, etcd Team"
+  - "@fzdarsky"
+  - "@deads2k"
+  - "@derekwaynecarr"
+  - "@mangelajo"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2022-09-16
+last-updated: 2022-10-19
+tracking-link:
+  - https://issues.redhat.com/browse/ETCD-318
+#see-also:
+#  - "/enhancements/this-other-neat-thing.md"
+---
+
+# etcd as a Transient systemd Unit
+
+## Summary
+
+The enhancement proposes moving etcd out of the MicroShift binary and into a transient systemd unit that is launched and managed by MicroShift.
+
+## Motivation
+
+Currently MicroShift [bundles upstream etcd](https://github.com/openshift/microshift/blob/f7f2260e4fbff61654f478fc149cb7052261f87f/go.mod#L21) into its binary and runs it as a goroutine.
+This results in etcd dependencies being coupled with MicroShift which makes it harder to maintain and upgrade etcd independently of the platform like it is done with a separate openshift/etcd repo in the Openshift Container Platform (OCP).
+
+This enhancement outlines an update to MicroShift to change how etcd is bundled and deployed with MicroShift in order to minimize risk of shared dependencies between etcd and OpenShift/Kubernetes leading to hard-to-trace bugs.
+
+Changing the deployment of etcd, outlined below, would alleviate concerns about building etcd with a version of shared dependencies - mainly grpc - that etcd is not currently built against, and therefore, not tested against.
+
+Additionally moving the etcd deployment into its own process would make it easier to debug and process logs via journatlctl.
+
+### User Stories
+
+1. "As an etcd developer, I want to maintain etcd versions independently of microshift dependencies."
+
+2. "As an admin, I want to observe and debug the etcd server logs separately from other microshift processes."
+
+### Goals
+
+- The etcd binary is built and shipped separately from the microshift binary to ease maintainence.
+- The etcd server logs are easier to observe.
+
+### Non-Goals
+
+- Certificate management for etcd client authentication needs to be updated regardless of the delivery/execution implementation so this will be out of scope for this enhancement.
+- Creating an RPM build root specifically for MicroShift-etcd; this should be eventually needed so we're building etcd with the Golang version it's expecting (1.16), instead of the same that MicroShift is built with (1.19).
+  - This enhancement discusses this further in the 'etcd Binary from an RPM' section.
+  - This work should be done in the future, but is out of scope for this enhancement.
+
+## Proposal
```

Comment:
Let's clean this section up so it only includes the final proposal. The rest of the approaches should move down to alternatives and include a summary of why we rejected them.